src/autopas/selectors/tuningStrategy/PredictiveTuning.h

```diff
@@ -0,0 +1,370 @@
+  }
+
+  auto optimum = std::min_element(_traversalPredictions.begin(), _traversalPredictions.end(),
+                                  [](std::pair<Configuration, size_t> a, std::pair<Configuration, size_t> b) -> bool {
+                                    return a.second < b.second;
+                                  });
+
+  if (_validSearchSpace.count(optimum->first) == 0) {
+    autopas::utils::ExceptionHandler::exception("Predictive Tuning: No valid optimal configuration could be found");
+  }
+
+  _optimalSearchSpace.emplace(optimum->first);
+
+  // selects configurations that are near the optimal prediction
+  for (auto &configuration : _validSearchSpace) {
+    if ((float)_traversalPredictions[configuration] / optimum->second <= 1.2) {
+      _optimalSearchSpace.emplace(configuration);
+    }
+  }
+
+  // sanity check
+  if (_optimalSearchSpace.empty()) {
+    autopas::utils::ExceptionHandler::exception("PredicitveTuning: No possible configuration prediction found!");
+  }
+
+  _currentConfig = _searchSpace.begin();
+  while (_optimalSearchSpace.count(*_currentConfig) == 0 && _currentConfig != _searchSpace.end()) {
+    ++_currentConfig;
+  }
+}
+
+bool PredictiveTuning::tune(bool valid) {
+  if (!valid) {
+    _validConfiguration = true;
+  }
+
+  // repeat as long as traversals are not applicable or we run out of configs
+  do {
+    ++_currentConfig;
+  } while (_optimalSearchSpace.count(*_currentConfig) == 0 && _currentConfig != _searchSpace.end());
+
+  if (_currentConfig == _searchSpace.end()) {
+    if (_validConfiguration) {
+      selectOptimalConfiguration();
+      _timer++;
+      return false;
+    } else {
+      invalidOptimalSearchSpace();
+    }
+  }
+
+  return true;
+}
+
+void PredictiveTuning::selectOptimalConfiguration() {
+  if (_optimalSearchSpace.size() == 1) {
+    _currentConfig = _optimalSearchSpace.begin();
+    return;
+  }
+
+  // Time measure strategy
+  if (_traversalTimes.empty()) {
+    utils::ExceptionHandler::exception(
+        "PredictiveTuning: Trying to determine fastest configuration without any measurements! "
+        "Either selectOptimalConfiguration was called too early or no applicable configurations were found");
+  }
+
+  auto optimum = std::min_element(_traversalTimes.begin(), _traversalTimes.end(),
+                                  [](std::pair<Configuration, size_t> a, std::pair<Configuration, size_t> b) -> bool {
+                                    return a.second < b.second;
+                                  });
+
+  _currentConfig = _searchSpace.find(optimum->first);
+  // sanity check
+  if (_currentConfig == _searchSpace.end()) {
+    autopas::utils::ExceptionHandler::exception(
+        "PredicitveTuning: Optimal configuration not found in list of configurations!");
+  }
+
+  // measurements are not needed anymore
+  _traversalTimes.clear();
+  _traversalPredictions.clear();
+  _optimalSearchSpace.clear();
```

Comment:
true, but isn't this already handled by `reset()`? (I would probably have the same comment in `FullSearch()` so please check and fix this there too if applicable).